packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -162,17 +166,41 @@ export const App = ({
   // Can be looked at in FXA-7626 or FXA-7184
   return (
     <Router basepath="/">
-      <AuthAndAccountSetupRoutes {...{ isSignedIn }} path="/*" />
-      <SettingsRoutes {...{ isSignedIn }} path="/settings/*" />
+      <AuthAndAccountSetupRoutes {...{ isSignedIn, integration }} path="/*" />
+      <SettingsRoutes {...{ isSignedIn, integration }} path="/settings/*" />
     </Router>
   );
 };
 
 const SettingsRoutes = ({
   isSignedIn,
-}: { isSignedIn: boolean } & RouteComponentProps) => {
+  integration,
+}: { isSignedIn: boolean; integration: Integration } & RouteComponentProps) => {
   const location = useLocation();
-  if (isSignedIn === false) {
+  // TODO: Remove this + config.sendFxAStatusOnSettings check once we confirm this works
+  const config = useConfig();
+
+  // Request and update account data/state to match the browser state because:
+  // 1) If a user logs in through the browser Sync pairing flow and then accesses
+  // Settings through the browser, they'd have to login again (isSignedIn === false).
+  // 2) If a user disconnects their account through Sync and then logs into another account
+  // using the pairing flow, we hold onto the old session token (isSignedIn === true).
+  const [shouldCheckFxaStatus, setShouldCheckFxaStatus] = useState(
+    // Note we check the context here and not `isSync` because this is for a special case
+    // when mobile accesses Settings through the browser's "Manage account" (Web integration)
+    integration.data.context === Constants.OAUTH_WEBCHANNEL_CONTEXT &&
```

Comment:
I originally had this as `isOAuthIntegration(integration) && integration.isSync()` but Tarik called out that this isn't the OAuth flow (e.g. `client_id` is not provided), so technically _just_ `context=oauth_webchannel_v1` is considered a Web integration.

I thought about adding `isSync()` on the Web integration but I'm not sure if we want to consider that the "sync flow" because it's essentially just the browser loading Settings in web-view... though maybe that makes the most sense since we are "syncing" data between FxA and the browser? I feel like this is OK for now, but happy to hear other thoughts, it's easy enough to tweak.